File: 08-DesignPatterns-Java/Solution/Adapter/src/Quack/TurkeyQuackStrategy.java
package Quack;

import Turkeys.Turkey;

public final class TurkeyQuackStrategy implements IQuackStrategy {
    private final Turkeys.Turkey _turkey;

    public TurkeyQuackStrategy(Turkey turkey) {
        _turkey = turkey;
    }

    @Override
    public void quack() {
        _turkey.gobble();
    }
}

File: 08-DesignPatterns-CSharp/08-2-2-Adapter-Solution/Quack/TurkeyQuackStrategy.cs
namespace Adapter.Quack
{
    using Adapter.Turkeys;

    internal class TurkeyQuackStrategy : IQuackStrategy
    {
        private readonly Turkey _turkey;

        internal TurkeyQuackStrategy(Turkey turkey)
        {
            _turkey = turkey;
        }

        public void Quack() => _turkey.Gobble();
    }
}